components/widgets/menu/menu_items/submenu_item.jsx

```diff
@@ -0,0 +1,168 @@
+        const submenuModalData = {
+            ModalId: ModalIdentifiers.MOBILE_SUBMENU,
+            dialogType: SubMenuModal,
+            dialogProps: {
+                elements,
+                action,
+            },
+        };
+
+        store.dispatch(openModal(submenuModalData));
+    }
+
+    onClick = (event) => {
+        const {id, subMenu, action, root} = this.props;
+        const isMobile = Utils.isMobile();
+        if (isMobile) {
+            if (subMenu && subMenu.length) { // if contains a submenu, call openModal with it
+                if (!root) { //required to close only the original menu
+                    event.stopPropagation();
+                }
+                this.handleShowMobileSubMenuItem(subMenu, action);
+            } else { // leaf node in the tree handles action only
+                action(id);
+            }
+        } else if (event.nativeEvent.path && // the first 2 elements in path match original event id
+            event.nativeEvent.path.slice(0, 2).find((e) => e.id === id)) {
+            action(id);
+        } else if (!event.nativeEvent.path) { //for tests only that don't contain `path`
+            action(id);
+        }
+    }
+
+    render() {
+        const {id, text, subMenu, icon, action, xOffset, ariaLabel} = this.props;
+        const isMobile = Utils.isMobile();
+
+        let textProp = text;
+        if (icon) {
+            textProp = (
+                <React.Fragment>
+                    <span className='icon'>{icon}</span>
+                    {text}
+                </React.Fragment>
+            );
+        }
+
+        const hasSubmenu = subMenu && subMenu.length;
+        const parentWidth = this.node && this.node.current ? this.node.current.getBoundingClientRect().width - 19 : 0;
+        const subMenuStyle = {
+            visibility: this.state.show && hasSubmenu && !isMobile ? 'visible' : 'hidden',
+            right: (parseInt(xOffset, 10) - 1) + 'px',
+        };
+
+        let subMenuContent = '';
+
+        if (!isMobile) {
+            subMenuContent = (
+                <ul
+                    className={'a11y__popup Menu dropdown-menu SubMenu'}
+                    style={subMenuStyle}
+                >
+                    {isMobile || !subMenu ? '' : subMenu.map((s) => {
+                        return (
+                            <SubMenuItem
+                                key={s.id}
+                                id={s.id}
+                                text={s.text}
+                                subMenu={s.subMenu}
+                                action={action}
+                                xOffset={parentWidth}
+                                aria-label={ariaLabel}
+                                root={false}
+                            />
+                        );
+                    })}
+                </ul>
+            );
+        }
+
+        return (
+            <li
+                className='MenuItem'
```

Comment:
To follow the pattern used with other widgets, we should add here 'SubMenuItem'  class.

Reply:
added class